src/manage_nsfs/nc_lifecycle.js

```diff
@@ -60,11 +62,57 @@ async function throw_if_noobaa_not_active(config_fs, system_json) {
     }
 }
 
+function _get_file_age_days(stat) {
+    //TODO how much do we care about rounding errors? (it is by days after all)
+    return (Date.now() - Number(stat.mtimeNsBigint) / 1e6) / 24 / 60 / 60 / 1000;
+}
+
+function _list_contain_tag(query_tag, tag_set) {
+    for (const t of tag_set) {
+        if (t.key === query_tag.key && t.value === query_tag.value) return true;
+    }
+    return false;
+}
+
+function _file_contain_tags(object_info, filter_tags) {
+    if (object_info.tags === undefined) return false;
+    for (const tag of filter_tags) {
+        if (!_list_contain_tag(tag, object_info.tags)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @param {*} create_params_parsed
+ * @param {nb.NativeFSStats} stat
+ */
+function _get_lifecycle_object_info_for_mpu(create_params_parsed, stat) {
+    return {
+        key: create_params_parsed.key,
+        age: _get_file_age_days(stat),
+        tags: create_params_parsed.tagging,
+    };
+}
+
+function _build_lifecycle_filter(params) {
+    /**
+     * @param {*} object_info
+     */
+    return function(object_info) {
+        if (params.filter?.prefix && !object_info.key.startsWith(params.filter.prefix)) return false;
+        if (params.expiration && object_info.age < params.expiration) return false;
+        if (params.filter?.tags && !_file_contain_tags(object_info, params.tags)) return false;
```

Comment:
missing size filter IIUC

Reply:
its because its not needed for abort mpu. I will add it